touch/resistive/fourwire.go

```diff
@@ -0,0 +1,82 @@
+package resistive
+
+import (
+	"machine"
+
+	"tinygo.org/x/drivers/touch"
+)
+
+type FourWireTouchscreen struct {
+	YP machine.Pin
+	YM machine.Pin
+	XP machine.Pin
+	XM machine.Pin
+
+	yp machine.ADC
+	xm machine.ADC
+	xp machine.ADC
+
+	samples []uint16
+}
+
+func (res *FourWireTouchscreen) Configure() {
+	res.yp = machine.ADC{res.YP}
+	res.xm = machine.ADC{res.XM}
+	res.xp = machine.ADC{res.XP}
+	res.samples = make([]uint16, 2)
+}
+
+func (res *FourWireTouchscreen) GetTouchPoint() (p touch.Point) {
+	p.X = int(res.ReadX())
+	p.Y = int(res.ReadY())
+	p.Z = int(res.ReadZ())
+	return
+}
+
+func (res *FourWireTouchscreen) ReadX() uint16 {
+	res.YM.Configure(machine.PinConfig{machine.PinInput})
+	res.YM.Low()
```

Comment:
You cannot normally set a pin high or low when it's an input. What is the intention here?

Reply:
This is addressed in the latest set of commits.